File: src/reverse-integer.java
class Solution {
    public int reverse(int x) {
        if(x / 10 == 0) return x;
        boolean isNegative = false;
        if(x < 0){
            isNegative = true;
            x = x * -1;
        }
        int solution = 0;
        Stack<Integer> stack = new Stack<Integer>();
        int result = 1;
        int number = x;
        while(x /(result * 10) != 0){
            result *= 10;
        }
        int marker = result;
        while(result > 0){
            stack.push(number / result);
            number -= (number/result) * result;
            result /= 10;
        }
      
        while(!stack.empty()){
            solution += stack.pop() * marker;
            marker /= 10;
        }
        if(isNegative){
            solution *= -1;
        }
        return solution;
    }
}

File: src/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        result = 0
        isNegative = False
        if x < 0:
            isNegative = True
            x = x * -1
        while x != 0:
            tail = int(x) % 10
            newResult = int(result * 10 + tail)
            if (int((newResult - tail) / 10) != result): return 0
            result = newResult
            x = int(x / 10)
        if isNegative:
            result = result * -1
        return result